File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        # 중복 요소를 모두 제거하기
        s = set(nums)

        idx = 0
        non_duplicate_nums = list(s)
        non_duplicate_nums.sort() # O(NlogN)
        for x in non_duplicate_nums: # O(N)
            nums[idx] = x
            idx += 1
            
        return len(s)

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        
        HashSet<Integer> s = new HashSet<>();
        for (int num : nums) {
            s.add(num);
        }
        
        ArrayList<Integer> arr_nums = new ArrayList<>(s);
        Collections.sort(arr_nums);
        
        
        for (int i = 0; i < arr_nums.size(); i++) {
            nums[i] = arr_nums.get(i);
        }
        
        return arr_nums.size();
    }
}